patching/plugin-commands-patching/src/getPatchedDependency.ts

```diff
@@ -60,7 +60,7 @@ export async function getPatchedDependency (rawDependency: string, opts: GetPatc
     const preferred = preferredVersions[0]
     return {
       ...dep,
-      applyToAll: !dep.bareSpecifier,
+      applyToAll: !dep.bareSpecifier && !preferred.gitTarballUrl,
       bareSpecifier: preferred.gitTarballUrl ?? preferred.version,
     }
```

Comment:
now that  preferred.gitTarballUrl is used in both fields, it would look better with an if statement

```ts
if (preferred.gitTarballUrl) {
  return {
    ...opts,
    applyToAll: false,
    bareSpecifier: preferred.gitTarballUrl,
  }
}
return {
  ...opts,
  applyToAll: !dep.bareSpecifier,
  bareSpecifier: preferred.version,
}
```